Error while running "pip install -r requirements.txt"

Hello, 
I am trying to do the setup for using PointNet++. My environment contains Python = 3.6.10, cuda=10.1 and pytorch=1.5.0. If I try running "pip install -r requirements.txt" I receive the following errors during the compiling:

`ERROR: Command errored out with exit status 1:
   command: /home/alberto/anaconda3/envs/pn2/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-ezjt_ypu/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-ezjt_ypu/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-004bspdj
       cwd: /tmp/pip-req-build-ezjt_ypu/
  Complete output (169 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/pointnet2_ops
  copying pointnet2_ops/__init__.py -> build/lib.linux-x86_64-3.6/pointnet2_ops
  copying pointnet2_ops/pointnet2_utils.py -> build/lib.linux-x86_64-3.6/pointnet2_ops
  copying pointnet2_ops/_version.py -> build/lib.linux-x86_64-3.6/pointnet2_ops
  copying pointnet2_ops/pointnet2_modules.py -> build/lib.linux-x86_64-3.6/pointnet2_ops
  running egg_info
  creating pointnet2_ops.egg-info
  writing pointnet2_ops.egg-info/PKG-INFO
  writing dependency_links to pointnet2_ops.egg-info/dependency_links.txt
  writing top-level names to pointnet2_ops.egg-info/top_level.txt
  writing manifest file 'pointnet2_ops.egg-info/SOURCES.txt'
  reading manifest file 'pointnet2_ops.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'pointnet2_ops.egg-info/SOURCES.txt'
  creating build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src
  creating build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/ball_query.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/cuda_utils.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/group_points.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/interpolate.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/sampling.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  copying pointnet2_ops/_ext-src/include/utils.h -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/include
  creating build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/ball_query.cpp -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/ball_query_gpu.cu -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/bindings.cpp -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/group_points.cpp -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/group_points_gpu.cu -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/interpolate.cpp -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/interpolate_gpu.cu -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/sampling.cpp -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  copying pointnet2_ops/_ext-src/src/sampling_gpu.cu -> build/lib.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  running build_ext
  building 'pointnet2_ops._ext' extension
  creating /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6
  creating /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops
  creating /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src
  creating /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src
  Emitting ninja build file /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/build.ninja...
  Compiling objects...
  Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
  [1/9] c++ -MMD -MF /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src/sampling.o.d -pthread -B /home/alberto/anaconda3/envs/pn2/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ipointnet2_ops/_ext-src/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/TH -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/home/alberto/anaconda3/envs/pn2/include/python3.6m -c -c /tmp/pip-req-build-ezjt_ypu/pointnet2_ops/_ext-src/src/sampling.cpp -o /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src/sampling.o -O3 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
  FAILED: /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src/sampling.o
  c++ -MMD -MF /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src/sampling.o.d -pthread -B /home/alberto/anaconda3/envs/pn2/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ipointnet2_ops/_ext-src/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/TH -I/home/alberto/anaconda3/envs/pn2/lib/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/home/alberto/anaconda3/envs/pn2/include/python3.6m -c -c /tmp/pip-req-build-ezjt_ypu/pointnet2_ops/_ext-src/src/sampling.cpp -o /tmp/pip-req-build-ezjt_ypu/build/temp.linux-x86_64-3.6/pointnet2_ops/_ext-src/src/sampling.o -O3 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  /tmp/pip-req-build-ezjt_ypu/pointnet2_ops/_ext-src/src/sampling.cpp:1:10: fatal error: sampling.h: File o directory non esistente
   #include "sampling.h"
            ^~~~~~~~~~~~
`

It does the same thing for all the other files to be compiled. Can you please help with that?

--- a/README.rst
+++ b/README.rst
@@ -17,7 +17,10 @@ The custom ops used by Pointnet++ are currently **ONLY** supported on the GPU us
 Setup
 -----
 
-* Install ``python`` -- This repo is tested with ``3.6``
+* Install ``python`` -- This repo is tested with ``{3.6, 3.7}``
+
+* Install ``pytorch`` with CUDA -- This repo is tested with ``{1.4, 1.5}``.
+  It may work with versions newer than ``1.5``, but this is not guaranteed.
 
 
 * Install dependencies

--- a/pointnet2_ops_lib/pointnet2_ops/_ext-src/include/utils.h
+++ b/pointnet2_ops_lib/pointnet2_ops/_ext-src/include/utils.h
@@ -2,24 +2,24 @@
 #include <ATen/cuda/CUDAContext.h>
 #include <torch/extension.h>
 
-#define CHECK_CUDA(x)                                          \
-  do {                                                         \
-    AT_CHECK(x.type().is_cuda(), #x " must be a CUDA tensor"); \
+#define CHECK_CUDA(x)                                    \
+  do {                                                   \
+    AT_ASSERT(x.is_cuda(), #x " must be a CUDA tensor"); \
   } while (0)
 
-#define CHECK_CONTIGUOUS(x)                                         \
-  do {                                                              \
-    AT_CHECK(x.is_contiguous(), #x " must be a contiguous tensor"); \
+#define CHECK_CONTIGUOUS(x)                                          \
+  do {                                                               \
+    AT_ASSERT(x.is_contiguous(), #x " must be a contiguous tensor"); \
   } while (0)
 
-#define CHECK_IS_INT(x)                              \
-  do {                                               \
-    AT_CHECK(x.scalar_type() == at::ScalarType::Int, \
-             #x " must be an int tensor");           \
+#define CHECK_IS_INT(x)                               \
+  do {                                                \
+    AT_ASSERT(x.scalar_type() == at::ScalarType::Int, \
+              #x " must be an int tensor");           \
   } while (0)
 
-#define CHECK_IS_FLOAT(x)                              \
-  do {                                                 \
-    AT_CHECK(x.scalar_type() == at::ScalarType::Float, \
-             #x " must be a float tensor");            \
+#define CHECK_IS_FLOAT(x)                               \
+  do {                                                  \
+    AT_ASSERT(x.scalar_type() == at::ScalarType::Float, \
+              #x " must be a float tensor");            \
   } while (0)

--- a/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/ball_query.cpp
+++ b/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/ball_query.cpp
@@ -12,7 +12,7 @@ at::Tensor ball_query(at::Tensor new_xyz, at::Tensor xyz, const float radius,
   CHECK_IS_FLOAT(new_xyz);
   CHECK_IS_FLOAT(xyz);
 
-  if (new_xyz.type().is_cuda()) {
+  if (new_xyz.is_cuda()) {
     CHECK_CUDA(xyz);
   }
 
@@ -20,12 +20,12 @@ at::Tensor ball_query(at::Tensor new_xyz, at::Tensor xyz, const float radius,
       torch::zeros({new_xyz.size(0), new_xyz.size(1), nsample},
                    at::device(new_xyz.device()).dtype(at::ScalarType::Int));
 
-  if (new_xyz.type().is_cuda()) {
+  if (new_xyz.is_cuda()) {
     query_ball_point_kernel_wrapper(xyz.size(0), xyz.size(1), new_xyz.size(1),
                                     radius, nsample, new_xyz.data_ptr<float>(),
                                     xyz.data_ptr<float>(), idx.data_ptr<int>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return idx;

--- a/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/group_points.cpp
+++ b/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/group_points.cpp
@@ -15,7 +15,7 @@ at::Tensor group_points(at::Tensor points, at::Tensor idx) {
   CHECK_IS_FLOAT(points);
   CHECK_IS_INT(idx);
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     CHECK_CUDA(idx);
   }
 
@@ -23,13 +23,13 @@ at::Tensor group_points(at::Tensor points, at::Tensor idx) {
       torch::zeros({points.size(0), points.size(1), idx.size(1), idx.size(2)},
                    at::device(points.device()).dtype(at::ScalarType::Float));
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     group_points_kernel_wrapper(points.size(0), points.size(1), points.size(2),
                                 idx.size(1), idx.size(2),
                                 points.data_ptr<float>(), idx.data_ptr<int>(),
                                 output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;
@@ -41,7 +41,7 @@ at::Tensor group_points_grad(at::Tensor grad_out, at::Tensor idx, const int n) {
   CHECK_IS_FLOAT(grad_out);
   CHECK_IS_INT(idx);
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     CHECK_CUDA(idx);
   }
 
@@ -49,13 +49,13 @@ at::Tensor group_points_grad(at::Tensor grad_out, at::Tensor idx, const int n) {
       torch::zeros({grad_out.size(0), grad_out.size(1), n},
                    at::device(grad_out.device()).dtype(at::ScalarType::Float));
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     group_points_grad_kernel_wrapper(
         grad_out.size(0), grad_out.size(1), n, idx.size(1), idx.size(2),
         grad_out.data_ptr<float>(), idx.data_ptr<int>(),
         output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;

--- a/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/interpolate.cpp
+++ b/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/interpolate.cpp
@@ -17,7 +17,7 @@ std::vector<at::Tensor> three_nn(at::Tensor unknowns, at::Tensor knows) {
   CHECK_IS_FLOAT(unknowns);
   CHECK_IS_FLOAT(knows);
 
-  if (unknowns.type().is_cuda()) {
+  if (unknowns.is_cuda()) {
     CHECK_CUDA(knows);
   }
 
@@ -28,12 +28,12 @@ std::vector<at::Tensor> three_nn(at::Tensor unknowns, at::Tensor knows) {
       torch::zeros({unknowns.size(0), unknowns.size(1), 3},
                    at::device(unknowns.device()).dtype(at::ScalarType::Float));
 
-  if (unknowns.type().is_cuda()) {
+  if (unknowns.is_cuda()) {
     three_nn_kernel_wrapper(unknowns.size(0), unknowns.size(1), knows.size(1),
                             unknowns.data_ptr<float>(), knows.data_ptr<float>(),
                             dist2.data_ptr<float>(), idx.data_ptr<int>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return {dist2, idx};
@@ -48,7 +48,7 @@ at::Tensor three_interpolate(at::Tensor points, at::Tensor idx,
   CHECK_IS_INT(idx);
   CHECK_IS_FLOAT(weight);
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     CHECK_CUDA(idx);
     CHECK_CUDA(weight);
   }
@@ -57,13 +57,13 @@ at::Tensor three_interpolate(at::Tensor points, at::Tensor idx,
       torch::zeros({points.size(0), points.size(1), idx.size(1)},
                    at::device(points.device()).dtype(at::ScalarType::Float));
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     three_interpolate_kernel_wrapper(
         points.size(0), points.size(1), points.size(2), idx.size(1),
         points.data_ptr<float>(), idx.data_ptr<int>(), weight.data_ptr<float>(),
         output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;
@@ -77,7 +77,7 @@ at::Tensor three_interpolate_grad(at::Tensor grad_out, at::Tensor idx,
   CHECK_IS_INT(idx);
   CHECK_IS_FLOAT(weight);
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     CHECK_CUDA(idx);
     CHECK_CUDA(weight);
   }
@@ -86,13 +86,13 @@ at::Tensor three_interpolate_grad(at::Tensor grad_out, at::Tensor idx,
       torch::zeros({grad_out.size(0), grad_out.size(1), m},
                    at::device(grad_out.device()).dtype(at::ScalarType::Float));
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     three_interpolate_grad_kernel_wrapper(
         grad_out.size(0), grad_out.size(1), grad_out.size(2), m,
         grad_out.data_ptr<float>(), idx.data_ptr<int>(),
         weight.data_ptr<float>(), output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;

--- a/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/sampling.cpp
+++ b/pointnet2_ops_lib/pointnet2_ops/_ext-src/src/sampling.cpp
@@ -18,7 +18,7 @@ at::Tensor gather_points(at::Tensor points, at::Tensor idx) {
   CHECK_IS_FLOAT(points);
   CHECK_IS_INT(idx);
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     CHECK_CUDA(idx);
   }
 
@@ -26,12 +26,12 @@ at::Tensor gather_points(at::Tensor points, at::Tensor idx) {
       torch::zeros({points.size(0), points.size(1), idx.size(1)},
                    at::device(points.device()).dtype(at::ScalarType::Float));
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     gather_points_kernel_wrapper(points.size(0), points.size(1), points.size(2),
                                  idx.size(1), points.data_ptr<float>(),
                                  idx.data_ptr<int>(), output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;
@@ -44,7 +44,7 @@ at::Tensor gather_points_grad(at::Tensor grad_out, at::Tensor idx,
   CHECK_IS_FLOAT(grad_out);
   CHECK_IS_INT(idx);
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     CHECK_CUDA(idx);
   }
 
@@ -52,13 +52,13 @@ at::Tensor gather_points_grad(at::Tensor grad_out, at::Tensor idx,
       torch::zeros({grad_out.size(0), grad_out.size(1), n},
                    at::device(grad_out.device()).dtype(at::ScalarType::Float));
 
-  if (grad_out.type().is_cuda()) {
+  if (grad_out.is_cuda()) {
     gather_points_grad_kernel_wrapper(grad_out.size(0), grad_out.size(1), n,
                                       idx.size(1), grad_out.data_ptr<float>(),
                                       idx.data_ptr<int>(),
                                       output.data_ptr<float>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;
@@ -75,12 +75,12 @@ at::Tensor furthest_point_sampling(at::Tensor points, const int nsamples) {
       torch::full({points.size(0), points.size(1)}, 1e10,
                   at::device(points.device()).dtype(at::ScalarType::Float));
 
-  if (points.type().is_cuda()) {
+  if (points.is_cuda()) {
     furthest_point_sampling_kernel_wrapper(
         points.size(0), points.size(1), nsamples, points.data_ptr<float>(),
         tmp.data_ptr<float>(), output.data_ptr<int>());
   } else {
-    AT_CHECK(false, "CPU not supported");
+    AT_ASSERT(false, "CPU not supported");
   }
 
   return output;

--- a/pointnet2_ops_lib/setup.py
+++ b/pointnet2_ops_lib/setup.py
@@ -5,13 +5,14 @@ import os.path as osp
 from setuptools import find_packages, setup
 from torch.utils.cpp_extension import BuildExtension, CUDAExtension
 
+this_dir = osp.dirname(osp.abspath(__file__))
 _ext_src_root = osp.join("pointnet2_ops", "_ext-src")
 _ext_sources = glob.glob(osp.join(_ext_src_root, "src", "*.cpp")) + glob.glob(
     osp.join(_ext_src_root, "src", "*.cu")
 )
 _ext_headers = glob.glob(osp.join(_ext_src_root, "include", "*"))
 
-requirements = []
+requirements = ["torch>=1.4"]
 
 exec(open(osp.join("pointnet2_ops", "_version.py")).read())
 
@@ -30,7 +31,7 @@ setup(
                 "cxx": ["-O3"],
                 "nvcc": ["-O3", "-Xfatbin", "-compress-all"],
             },
-            include_dirs=[osp.join(_ext_src_root, "include")],
+            include_dirs=[osp.join(this_dir, _ext_src_root, "include")],
         )
     ],
     cmdclass={"build_ext": BuildExtension},

